File: sdks/java/src/main/java/bridgebank/api/bancos/BancosRequestBuilder.java
package bridgebank.api.bancos;

import bridgebank.api.bancos.geradores.GeradoresRequestBuilder;
import bridgebank.api.bancos.parsers.ParsersRequestBuilder;
import com.microsoft.kiota.BaseRequestBuilder;
import com.microsoft.kiota.RequestAdapter;
import java.util.HashMap;
import java.util.Objects;
/**
 * Builds and executes requests for operations under /api/bancos
 */
@jakarta.annotation.Generated("com.microsoft.kiota")
public class BancosRequestBuilder extends BaseRequestBuilder {
    /**
     * The geradores property
     * @return a {@link GeradoresRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public GeradoresRequestBuilder geradores() {
        return new GeradoresRequestBuilder(pathParameters, requestAdapter);
    }
    /**
     * The parsers property
     * @return a {@link ParsersRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public ParsersRequestBuilder parsers() {
        return new ParsersRequestBuilder(pathParameters, requestAdapter);
    }
    /**
     * Instantiates a new {@link BancosRequestBuilder} and sets the default values.
     * @param pathParameters Path parameters for the request
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public BancosRequestBuilder(@jakarta.annotation.Nonnull final HashMap<String, Object> pathParameters, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/api/bancos", pathParameters);
    }
    /**
     * Instantiates a new {@link BancosRequestBuilder} and sets the default values.
     * @param rawUrl The raw URL to use for the request builder.
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public BancosRequestBuilder(@jakarta.annotation.Nonnull final String rawUrl, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/api/bancos", rawUrl);
    }
}

File: sdks/python/bridgebank/api/bancos/bancos_request_builder.py
from __future__ import annotations
from collections.abc import Callable
from kiota_abstractions.base_request_builder import BaseRequestBuilder
from kiota_abstractions.get_path_parameters import get_path_parameters
from kiota_abstractions.request_adapter import RequestAdapter
from typing import Any, Optional, TYPE_CHECKING, Union

if TYPE_CHECKING:
    from .geradores.geradores_request_builder import GeradoresRequestBuilder
    from .parsers.parsers_request_builder import ParsersRequestBuilder

class BancosRequestBuilder(BaseRequestBuilder):
    """
    Builds and executes requests for operations under /api/bancos
    """
    def __init__(self,request_adapter: RequestAdapter, path_parameters: Union[str, dict[str, Any]]) -> None:
        """
        Instantiates a new BancosRequestBuilder and sets the default values.
        param path_parameters: The raw url or the url-template parameters for the request.
        param request_adapter: The request adapter to use to execute the requests.
        Returns: None
        """
        super().__init__(request_adapter, "{+baseurl}/api/bancos", path_parameters)
    
    @property
    def geradores(self) -> GeradoresRequestBuilder:
        """
        The geradores property
        """
        from .geradores.geradores_request_builder import GeradoresRequestBuilder

        return GeradoresRequestBuilder(self.request_adapter, self.path_parameters)
    
    @property
    def parsers(self) -> ParsersRequestBuilder:
        """
        The parsers property
        """
        from .parsers.parsers_request_builder import ParsersRequestBuilder

        return ParsersRequestBuilder(self.request_adapter, self.path_parameters)
    

